src/middleware/authorisation.ts

```diff
@@ -0,0 +1,45 @@
+import { Request, Response, NextFunction } from "express";
+
+export function requireRole(requiredRole: string) {
+    return function (req: Request, res: Response, next: NextFunction) {
+        if (req.session && (req.session.user?.role == 'Admin' || req.session.user?.role === requiredRole)) {
+            next();
+        } else {
```

Comment:
You could remove the else and nested if by doing return next() and putting the logic underneath